### PR TITLE
Bug 1807307 - Disable LeakCanary in ui-test-apk with gradle flag

### PR DESCRIPTION
**Pull request by jonalmeida**
This is the alternative solution if https://github.com/mozilla-mobile/firefox-android/pull/3967 doesn't stick. It's not as nice as using the official way to disable LeakCanary, but it should work.

~~(Will add the bug number to the title if we go this route to have it auto-linked).~~

🗒️ When this lands, I will share this in our mailing lists in case folks are relying on LeakCanary being enabled in CI builds for some reason.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1807307

### CODE REVIEW

File: fenix/app/src/main/res/xml/preferences.xml [hunk 1]
```diff
@@ -166,7 +166,6 @@
             android:title="@string/preferences_external_download_manager" />
 
         <androidx.preference.SwitchPreference
-            android:defaultValue="true"
```
Comment (jonalmeida) [inline fenix/app/src/main/res/xml/preferences.xml:169]: It's hard to sync this default value with what we have in our preferences so it's best of we don't use the default here and let it read it from the pref instead.

**Comment by jonalmeida** [comment]
I verified locally and from the bytecode in the APK inspector that the fenix debug APK from `build-apk-fenix-debug` is now false, so this should be used in the `ui-test-apk` that is scheduled during `main` branch pushes.
Screenshot 2023-10-06 at 4 18 49 PM

**Comment by jonalmeida** [comment]
@AaronMT if there is any way to verify this before landing it in-tree, please let me know and I'd be happy to run those tests. Thanks!

This should be ready for review now (and landing on Monday).

**Comment by AaronMT** [comment]
If you want to give Robo 

> @AaronMT if there is any way to verify this before landing it in-tree, please let me know and I'd be happy to run those tests. Thanks!
> 
> This should be ready for review now (and landing on Monday).

If you'd like to try a Robo run here, you could add `github-pull-request` over in https://github.com/mozilla-mobile/firefox-android/blob/b0024e0ad972a0ca743932cfded950c63238e871/taskcluster/ci/ui-test-apk/kind.yml#L332 

and swap `1` with a handful of times if you'd like over in 
https://github.com/mozilla-mobile/firefox-android/blob/b0024e0ad972a0ca743932cfded950c63238e871/fenix/automation/taskcluster/androidTest/flank-arm-robo-test.yml#L37

**Comment by jonalmeida** [comment]
Thanks Aaron! I pushed a separate commit that I'll remove before landing.

**Review by AaronMT — APPROVED** [review]
LGTM

**Comment by jonalmeida** [comment]
Currently hitting BZ-1857872.

**Pull request merged**